Automattic/jetpack
# WooCommerce Analytics: Initialize the package only if is not loaded in WooCommerce Core

Fixes https://github.com/woocommerce/woocommerce/issues/45210
PT: pcShBQ-1Vt-p2

## Proposed changes:

* This PR initializes WooCommerce_Analytics package only if it was not loaded in WooCommerce core

To do that, we created a `WC_ANALYTICS` constant in the WooCommerce core version when we load the package. 
When this constant is present Jetpack will not initialize WooCommerce_Analytics.

This is for maintaining compatibility as it might happen that JEtpack gets updated but WooCommerce does not and vice versa. 

### Other information:


## Jetpack product discussion



PT: pcShBQ-1Vt-p2

## Does this pull request change what data or activity we track or use?



Added ✅ 

## Testing instructions:






* Connect Jetpack and install WooCommerce
* Go to the shop as guest (or not an admin), add a product and then go to the cart
* In the web source you will see `<script>window._wca = window._wca || [];</script>` script loaded 
* add the constant: `define( 'WC_ANALYTICS', true )`
* Go to the shop as guest (or not an admin), add a product and then go to the cart
* In the web source you won't see `<script>window._wca = window._wca || [];</script>` script loaded
